service-workers/service-worker/unregister-then-register-new-script.https.html

```diff
@@ -72,113 +72,54 @@
       .catch(unreached_rejection(t));
 }, 'Registering a new script URL while an unregistered registration is in use');
 
-async_test(function(t) {
-    var scope = 'resources/scope/unregister-then-register-new-script-that-404s';
-    var iframe;
-    var registration;
+promise_test(async function(t) {
+  const scope = 'resources/scope/unregister-then-register-new-script-that-404s';
+  const registration = await service_worker_unregister_and_register(t, worker_url, scope);
+  await wait_for_state(t, registration.installing, 'activated');
 
-    service_worker_unregister_and_register(t, worker_url, scope)
-      .then(function(r) {
-          registration = r;
-          return wait_for_state(t, r.installing, 'activated');
-        })
-      .then(function() {
-          return with_iframe(scope);
-        })
-      .then(function(frame) {
-          iframe = frame;
-          return registration.unregister();
-        })
-      .then(function() {
-          // Step 5.1 of Register clears the uninstall flag before fetching
-          // the script:
-          //
-          //  https://w3c.github.io/ServiceWorker/#register-algorithm
-          var promise = navigator.serviceWorker.register('this-will-404',
-                                                         { scope: scope });
-          return promise;
-        })
-      .then(
-        function() {
-          assert_unreached('register should reject the promise');
-        },
-        function() {
-          assert_equals(registration.installing, null,
-                        'registration.installing');
-          assert_equals(registration.waiting, null,
-                        'registration.waiting');
-          assert_equals(registration.active.scriptURL, normalizeURL(worker_url),
-                        'registration.active');
-          iframe.remove();
-          return with_iframe(scope);
-        })
-      .then(function(frame) {
-          assert_equals(
-              frame.contentWindow.navigator.serviceWorker.controller.scriptURL,
-              normalizeURL(worker_url),
-              'the original worker should control a new document');
-          frame.remove();
-          return registration.unregister();
-        })
-      .then(function() {
-          t.done();
-        })
-      .catch(unreached_rejection(t));
-}, 'Registering a new script URL that 404s does resurrect an ' +
-       'unregistered registration');
+  const iframe = await with_iframe(scope);
 
-async_test(function(t) {
-    var scope = 'resources/scope/unregister-then-register-reject-install-worker';
-    var iframe;
-    var registration;
+  await registration.unregister();
```

Comment:
To be super resilient you could additionally use a t.add_cleanup() function that removes the iframes and calls service_worker_unregister (or equivalent). That might actually be good as most browsers will initially fail this test.

Reply:
Ohhh I see `add_cleanup` supports async cleanup now. Excellent. I'll add that.